components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -73,65 +82,248 @@ public AbstractHTTPFunction() {
         allowedDomains = ConfigProvider.getInstance().getAllowedDomainsForHttpFunctions();
     }
 
-    protected void executeHttpMethod(HttpUriRequest request, Map<String, Object> eventHandlers) {
+    public enum RetryDecision {
+        RETRY,
+        NO_RETRY;
+
+        public boolean shouldRetry() {
+            return this == RETRY;
+        }
+    }
```

Comment:
This enum is used within this class context only. Do we need to make this public ?
I think we can make it a private one or a protected one so that this enum can be utilized in subclasses

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/0da007ebe0bbc0242efba6c485f5d4f3a3cb2fe2